### DISCUSSION TITLE
Triggering a URL on task completion

### DISCUSSION THREAD

**Discussion by polygami** [Ayagikei/LifeUp#123] [Q&A] [answered]
Hi,

I really love the app and have been using it for a few years now. It is my main app for tracking repeating habits and giving myself rewards. I do however want to create some integration with Obsidian, where completing a task in LifeUp would trigger an action in Obsidian. If I can trigger a MacroDroid URL on completion i.e. https://trigger.macrodroid.com/xxxx-xxxx/LifeUp?id=42, I can then then trigger an action in Obsidian via the advanced-uri plugin.

What I would like to know is, is it possible to trigger a URL on task completion? If not are there any other workarounds that I could try?

Thanks in advance.

**Comment by Ayagikei** [comment, marked as answer]
Hello, thanks for using LifeUp and trying to integrate it with other apps!

In-app solutions:
Maybe you can only achieve it indirectly by completing the task->reward items->using the item to open the URL.

API solution:
LifeUp provides broadcast events for completing tasks. You can use other automation tools such as Tasker, Macrodroid, or your own Android program to receive this event and complete subsequent integration.

The approximate implementation is:
1. LifeUp enables `Broadcast events`.
2. Use automation tools to monitor the completion event of a specific task.
3. Use automation tools to trigger the automatic opening of URLs.

For more information, please see: [https://docs.lifeupapp.fun/en/index.html#/guide/api?id=broadcast-event-notification](https://docs.lifeupapp.fun/en/index.html#/guide/api?id=broadcast-event-notification)

**Reply by polygami** [reply]
Thanks for the very quick response. After more research I actually came to that page and it solved everything.

I used `app.lifeup.task.complete` in the "Intent Received" trigger in MacroDroid, saved the id and name to local variables, then added the "Open Website" option to pass those parameters to Obsidian, using the [Advanced URI](https://github.com/Vinzent03/obsidian-advanced-uri/blob/master/README.md) plugin to trigger a [Templater](https://github.com/SilentVoid13/Templater) command.

It all worked incredibly smoothly in the end. Including the ability to listen for these events opens up an endless range of possibilities. I really love the app and being able to customise everything and dive into these kinds of automations is a lot of fun.
